include/tables/dmadata_table.h

```diff
@@ -0,0 +1,9 @@
+/**
+ * Select dmadata table for version
+ */
+#ifdef NON_MATCHING
+// For non-matching builds, dmadata is generated from the specfile segments
+#include "dmadata_table_spec.h"
+#else
```

Comment:
The spec is designed to match the order files appear in the rom. The ordering of files in dmadata is not the same (though is almost the same) as the ordering in rom, so by extension is not the same as the ordering in the spec. The only way around this that I can think of is to add more spec directives to define the dmadata ordering, which is what I originally thought to do, however after discussing in discord we thought it may be better to just hardcode the matching dmadata ordering rather than adding all this extra machinery. Something that also complicates matters somewhat is the buffers segment, since it doesn't exist in dmadata at all as it is only bss, this would need special handling as well if we wanted to generate a matching dmadata.